custom_components/tuya_local/devices/alecoair_d12_dehumidifier.yaml

```diff
@@ -117,14 +117,32 @@ secondary_entities:
           - dps_val: true
             icon: "mdi:hand-back-right-off"
   - entity: binary_sensor
-    name: Tank
     class: problem
-    category: diagnostic
+    name: Tank full
+    icon: "mdi:cup-water"
+    optional: true
+    dps:
+      - id: 19
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: true
+          - value: false
+  - entity: binary_sensor
+    class: problem
+    name: Fault
+    icon: "mdi:engine-off-outline"
```

Comment:
I don't think this icon is a substantial improvement over the default icon for problem sensors, so I think it is better to leave it off.